### PR TITLE
Fixes #25992 - pulp redirect for full disk fixed

### PR DESCRIPTION
**Pull request by lzap**
This fixes 302 response from pulp for ondemand sync repos.

### CODE REVIEW

**Comment by timogoebel** [comment]
@lzap: What do you think of switching to a proper file downloading library like https://github.com/janko/down ?

**Comment by lzap** [comment]
I feel like it's probably not necessary, we only have this place and it served well. It's no rocket science I guess.

**Comment by timogoebel** [comment]
@lzap: Where is the code that handles `limit`?

**Comment by lzap** [comment]
Whoops. As I said, this is not a rocket science ;-)

**Comment by timogoebel** [comment]
Looks good, but can we please add some tests for this?

**Comment by lzap** [comment]
Absolutely, good idea. Pulled webmock for that, ideal tool.  Also copied few more Rubocop rules from core which are bugging me.

File: test/unit/iso_generator_test.rb [hunk 1]
```diff
@@ -14,7 +14,20 @@ class IsoGeneratorTest < ActiveSupport::TestCase
       let(:template) { FactoryBot.create(:provisioning_template, template: 'Fake kernel line <%= @kernel %> - <%= @initrd %>') }
 
       setup do
-        host.expects(:provisioning_template).with(kind: :PXELinux).returns(template)
+        host.stubs(:provisioning_template).with(kind: :PXELinux).returns(template)
+      end
+
+      test 'fetch handles redirect' do
+        begin
```
Comment (timogoebel) [inline test/unit/iso_generator_test.rb]: You can instead use `Dir.mktmpdir do |dir|` to create a temporary dir that gets automatically deleted at the end of the block.

```ruby
Dir.mktmpdir do |dir|
  # ...
  ForemanBootdisk::ISOGenerator.fetch(File.join(dir, 'bootdisk-test-redirect'), url)
end

File: .rubocop.yml [hunk 2]
```diff
@@ -54,3 +36,10 @@ Style/ClassAndModuleChildren:
 Naming/FileName:
   Exclude:
     - 'db/seeds.d/**/*'
+# Support both ruby19 and hash_rockets
```
Comment (timogoebel) [inline .rubocop.yml]: nit: missing newline

File: .rubocop.yml [hunk 3]
```diff
@@ -14,8 +14,8 @@ Style/FormatStringToken:
 Style/DoubleNegation:
   Enabled: false
 
-Metrics/LineLength:
-  Max: 180
+Metrics:
+  Enabled: false
```
Comment (timogoebel) [inline .rubocop.yml]: ❤️

File: .rubocop.yml [hunk 4]
```diff
@@ -54,3 +36,10 @@ Style/ClassAndModuleChildren:
 Naming/FileName:
   Exclude:
     - 'db/seeds.d/**/*'
+# Support both ruby19 and hash_rockets
+Style/HashSyntax:
+  Enabled: false
```
Comment (timogoebel) [inline .rubocop.yml]: Can we change this to `EnforcedStyle: no_mixed_keys` so you have to at least decide for one or the other syntax?

**Comment by lzap** [comment]
All done!

File: .rubocop.yml (hunk 4 above)
Comment (timogoebel) [reply .rubocop.yml]: @lzap: This is still missing. ;-)

**Comment by lzap** [comment]
Fixed tests hopefully.

**Comment by timogoebel** [comment]
Thanks, @lzap.

**Pull request merged**